src/App.tsx

```diff
@@ -1,24 +1,237 @@
+
+  const activeTodos = todos.filter(({ completed }) => !completed);
+  const allCompleted = todos.filter(({ completed }) => completed);
+  const isAllCompleted = allCompleted.length === todos.length;
+
+  const pushError = (message: string) => {
+    setErrorMessage(message);
+    warningTimer(setErrorMessage, '', 3000);
+  };
+
+  const handleChangeEvent = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    setTitle(value);
+  };
+
+  const handleAddTodo = useCallback(async (e: React.FormEvent<HTMLFormElement>) => {
+    try {
+      e.preventDefault();
+
+      if (!title.trim().length) {
+        pushError('Title can\'t be empty');
+        setTitle('');
+
+        return;
+      }
+
+      const newTodo = {
+        id: 0,
+        title,
+        userId: USER_ID,
+        completed: false,
+      };
+
+      setCreatingTodo(newTodo);
+
+      const addedTodo = await createTodo(USER_ID, newTodo);
+
+      setTodos(currentTodos => [...currentTodos, addedTodo]);
+    } catch (error) {
+      pushError('Unable to add a todo');
+    } finally {
+      setCreatingTodo(null);
+      setTitle('');
+    }
+  }, [title]);
+
+  const onRemoveTodo = useCallback(async (removeTodo: Todo) => {
+    try {
+      setTodosInProcessed(currentTodos => [...currentTodos, removeTodo]);
+      await deleteTodo(USER_ID, removeTodo.id);
+
+      setTodos(prevTodos => prevTodos
+        .filter(({ id }) => id !== removeTodo.id));
+    } catch (error) {
+      pushError('Unable to delete a todo');
+    } finally {
+      setTodosInProcessed(currentTodos => currentTodos
+        .filter(({ id }) => id !== removeTodo.id));
+    }
+  }, []);
+
+  const clearCompleted = useCallback(() => {
+    allCompleted.forEach(async todo => {
+      try {
+        setTodosInProcessed(currentTodos => [...currentTodos, todo]);
+        await deleteTodo(USER_ID, todo.id);
+        setTodos(prevTodos => prevTodos.filter(({ completed }) => !completed));
+      } catch (error) {
+        pushError('Unable to delete a todo');
+      } finally {
+        setTodosInProcessed(currentTodos => currentTodos
+          .filter(({ completed }) => !completed));
+      }
+    });
+  }, [todos]);
+
+  const changeTodos = (todoChange: Todo) => {
+    setTodos(currentTodos => currentTodos.map(todo => {
+      return todo.id === todoChange.id
+        ? todoChange
+        : todo;
+    }));
+  };
+
+  const onToogleUpdateTodo = useCallback(async (todoTogle: Todo) => {
+    try {
+      setTodosInProcessed(currentTodos => [...currentTodos, todoTogle]);
+      const todoChangeStatus = await toogleTodo(USER_ID, todoTogle.id, !todoTogle.completed);
+
+      changeTodos(todoChangeStatus);
+    } catch (error) {
+      pushError('Unable to change completed');
+    } finally {
+      setTodosInProcessed(currentTodos => currentTodos
+        .filter(({ id }) => id !== todoTogle.id));
+    }
+  }, []);
+
+  const toogleUpdateTodos = useCallback(() => {
```

Comment:
```suggestion
  const toogleAllTodos = useCallback(() => {
```
This name will be clearer

Reply:
or even `onToggleAllTodos` to be consistent